src/shared/inc/dacprivate.h

```diff
@@ -811,10 +811,21 @@ struct MSLAYOUT DacpHeapSegmentData
         if (SUCCEEDED(hr))
         {
             // TODO:  This needs to be put on the Dac side.
-            if (this->segmentAddr == heap.generation_table[0].start_segment)
-                highAllocMark = heap.alloc_allocated;
+            if (heap.generation_table[0].start_segment != heap.generation_table[1].start_segment)
```

Comment:
hmm, this should be pushed down further into the DAC, basically QI and call something that encapsulates all this logic and falls back to the logic prior to this PR as needed. There you could ask `IsRegionGCEnabled` (and maybe that is also something that should be exposed as an API on the DAC for cases like this).

Reply:
Excuse my ignorance, but how exactly would I go about "pushing it further into the DAC"? I was trying to get by with a local change in SOS, but perhaps this isn't the right way to go? Keep in mind I only have the foggiest notions about DAC...

Reply:
The DAC is the runtime version specific component (in the runtime repo) that SOS calls (the GetHeapSegmentData call). Like Juan said it would be better to make the change in the runtime repo code but as long as this change doesn't break the SOS heap commands when targeting a 3.1 or 6.0 runtime, then I think it is ok.

Reply:
I see this needs more thought. I also realized that there are more places in SOS (eeheap.cpp mostly) that need changes for regions. I'll see if I can work something out.

Reply:
I worked out a fix on the DAC side here: https://github.com/dotnet/runtime/pull/69670.

I fixed more place to use seg.highAllocMark instead of seg.allocated.

I kept a small change in dacprivate.h to make sure a new SOS would keep working with an older DAC, and also that SOS would not step on the highAllocMark set in DAC.